tests/components/canary/conftest.py

```diff
@@ -0,0 +1,43 @@
+"""Define fixtures available for all tests."""
+from canary.api import Api
+from pytest import fixture
+
+from tests.async_mock import MagicMock, patch
+
+
+def mock_canary_update(self, **kwargs):
+    """Get the latest data from py-canary."""
+    for location in self._api.get_locations():
```

Comment:
Rather than duplicating the copy, maybe something like this in `canary/__init__.py` and patch `update` to `_update` in the test?


```
    @Throttle(MIN_TIME_BETWEEN_UPDATES)
    def update(self, **kwargs):
        """Get the latest data from py-canary."""
		return self._update(**kwargs)

    def _update(self, **kwargs):
        """Get the latest data from py-canary."""
        for location in self._api.get_locations():
            location_id = location.location_id

            self._locations_by_id[location_id] = location

            for device in location.devices:
                if device.is_online:
                    self._readings_by_device_id[
                        device.device_id
                    ] = self._api.get_latest_readings(device.device_id)
```